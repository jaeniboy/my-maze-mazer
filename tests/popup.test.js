import {describe, expect, test, vi} from "vitest"
import {generatePlayerListHTML, showPopup} from "../scripts/popup"
import {JSDOM} from "jsdom"

describe("popup window", () => {

    const dom = new JSDOM(`
        <div id="container">
            <div class="popup-overlay">
                <div class="popup-content">
                    <div class="close-button">
                    <div class="player-list-container"></div>
                </div>
            </div>
        </div>`)

    global.document = dom.window.document

    test("that players list renders correctly", () => {
        const fakePlayers = [
            {"name": "Luca Rossi", "time": "00:08.23"},
            {"name": "Sofia MÃ¼ller", "time": "00:07.89"},
            {"name": "Aisha Khan", "time": "00:09.56"},
            {"name": "Mateo Garcia", "time": "00:06.78"},
            {"name": "Yuki Tanaka", "time": "00:10.12"},
            {"name": "Anya Ivanova", "time": "00:07.45"},
            {"name": "Nina Schmidt", "time": "00:08.67"},
            {"name": "Omar Ahmed", "time": "00:09.34"}
        ]
        expect(generatePlayerListHTML(fakePlayers)).toMatchSnapshot()
    })

    test("that popup gets active", ()=>{
        vi.useFakeTimers()
        showPopup()
        vi.advanceTimersByTime(100)
        expect(document.querySelector(".popup-overlay").style.display).toBe("block")
        expect(document.querySelector(".popup-overlay").classList.contains("active")).toBeTruthy()
    },1200)
    
    test("that popup disappears on close", () => {
        document.querySelector(".close-button").click()
        vi.advanceTimersByTime(300)
        expect(document.querySelector(".popup-overlay").style.display).toBe("none")
        expect(document.querySelector(".popup-overlay").classList.contains("active")).toBeFalsy()
    })

    vi.useRealTimers()

})

