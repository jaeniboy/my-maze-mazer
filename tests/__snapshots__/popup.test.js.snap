// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`popup window > that players list renders correctly 1`] = `
<ul
  class="player-list"
>
  <li
    class="player-item"
  >
    <span
      class="player-name"
    >
      Luca Rossi
    </span>
    <span
      class="player-time"
    >
      00:08.23
    </span>
  </li>
  <li
    class="player-item"
  >
    <span
      class="player-name"
    >
      Sofia MÃ¼ller
    </span>
    <span
      class="player-time"
    >
      00:07.89
    </span>
  </li>
  <li
    class="player-item"
  >
    <span
      class="player-name"
    >
      Aisha Khan
    </span>
    <span
      class="player-time"
    >
      00:09.56
    </span>
  </li>
  <li
    class="player-item"
  >
    <span
      class="player-name"
    >
      Mateo Garcia
    </span>
    <span
      class="player-time"
    >
      00:06.78
    </span>
  </li>
  <li
    class="player-item"
  >
    <span
      class="player-name"
    >
      Yuki Tanaka
    </span>
    <span
      class="player-time"
    >
      00:10.12
    </span>
  </li>
  <li
    class="player-item"
  >
    <span
      class="player-name"
    >
      Anya Ivanova
    </span>
    <span
      class="player-time"
    >
      00:07.45
    </span>
  </li>
  <li
    class="player-item"
  >
    <span
      class="player-name"
    >
      Nina Schmidt
    </span>
    <span
      class="player-time"
    >
      00:08.67
    </span>
  </li>
  <li
    class="player-item"
  >
    <span
      class="player-name"
    >
      Omar Ahmed
    </span>
    <span
      class="player-time"
    >
      00:09.34
    </span>
  </li>
</ul>
`;
